core/rfb.js

```diff
@@ -1688,19 +1688,42 @@ RFB.messages = {
         buff[offset + 2] = 0; // padding
         buff[offset + 3] = 0; // padding
 
-        var n = text.length;
+        let length = text.length;
 
-        buff[offset + 4] = n >> 24;
-        buff[offset + 5] = n >> 16;
-        buff[offset + 6] = n >> 8;
-        buff[offset + 7] = n;
+        buff[offset + 4] = length >> 24;
+        buff[offset + 5] = length >> 16;
+        buff[offset + 6] = length >> 8;
+        buff[offset + 7] = length;
 
-        for (var i = 0; i < n; i++) {
-            buff[offset + 8 + i] =  text.charCodeAt(i);
-        }
+        sock._sQlen += 8;
 
-        sock._sQlen += 8 + n;
-        sock.flush();
+        // We can only flush a max of sQbufferSize bytes at a time and we have
+        // already used 8 bytes for the msg-type, padding and length.
+        let remaining = length;
+
+        // We have to keep track of from where in the text we begin creating the
+        // buffer for the flush in the next iteration.
+        let textOffset = 0;
+
+        while (remaining > 0) {
+
+            let flushSize = Math.min(remaining, (sock._sQbufferSize - sock._sQlen));
+            if (flushSize <= 0) {
+                break;
```

Comment:
Shouldn't we have a `fail()` in that case? We've screwed up the protocol at this point.

Reply:
yeah, I'll add that